### PR TITLE
feat(deployment): add init containers to pods

### LINKED ISSUE #28: Leverage initContainers for database migrations
Context: https://github.com/interledger/rafiki/issues/2621

### PR DESCRIPTION
**Pull request by golobitch**
## Overview
Just add initContainers for pod resource. Point of this initContainers is that they will run database migrations inside of this init containers.

Also did a little bit of formating of generated helm chart

## Context
fixes #28

### CODE REVIEW

**Comment by golobitch** [comment]
Build for this PR will fail with this error `npm ERR! Missing script: "knex -- migrate:latest --env production"` until this is not merged in the interledger/rafiki repository

File: charts/rafiki-auth/Chart.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.4.0
+version: 0.4.1
```
Comment (sabineschaller) [inline charts/rafiki-auth/Chart.yaml:20]: There is a version clash with your other PR but I'm sure you know

File: charts/rafiki-backend/templates/deployment.yaml [hunk 2]
```diff
@@ -57,21 +66,5 @@ spec:
             - name: connector
               containerPort: {{ .Values.port.connector }}
               protocol: TCP
-           #livenessProbe:
```
Comment (sabineschaller) [inline charts/rafiki-backend/templates/deployment.yaml:60]: Can you also remove that in `rafiki-frontend`?

**Review by sabineschaller — COMMENTED** [review]
I'm assuming we don't need this for frontend because there are no migrations there?

File: charts/rafiki-auth/Chart.yaml (hunk 1 above)
Comment (golobitch) [reply charts/rafiki-auth/Chart.yaml:20]: Yes I know. Thank you for pointing that out for me. I don't know which PR will be merged first, so I bumped patch version for both PR. When first one will be merged, the second one will get new patch number.

**Comment by golobitch** [comment]
> I'm assuming we don't need this for frontend because there are no migrations there?

Exactly

**Review by sabineschaller — APPROVED** [review]

**Comment by sabineschaller** [comment]
the workflow is failing with 
```
------------------------------------------------------------------------------------------------------------------------
 ✖︎ rafiki-auth => (version: "0.4.1", path: "charts/rafiki-auth") > failed waiting for process: exit status 1
 ✖︎ rafiki-backend => (version: "0.4.1", path: "charts/rafiki-backend") > failed waiting for process: exit status 1
 ✔︎ rafiki-frontend => (version: "0.4.1", path: "charts/rafiki-frontend")
------------------------------------------------------------------------------------------------------------------------
failed installing charts: failed processing charts
Error: Process completed with exit code 1.
```
I'm not sure that is the same from what you pointed out.

**Comment by golobitch** [comment]
> the workflow is failing with
> 
> ```
> ------------------------------------------------------------------------------------------------------------------------
>  ✖︎ rafiki-auth => (version: "0.4.1", path: "charts/rafiki-auth") > failed waiting for process: exit status 1
>  ✖︎ rafiki-backend => (version: "0.4.1", path: "charts/rafiki-backend") > failed waiting for process: exit status 1
>  ✔︎ rafiki-frontend => (version: "0.4.1", path: "charts/rafiki-frontend")
> ------------------------------------------------------------------------------------------------------------------------
> failed installing charts: failed processing charts
> Error: Process completed with exit code 1.
> ```
> 
> I'm not sure that is the same from what you pointed out.

Yes, we can see in failed action following line, that is actually output of k8s pod for rafiki-auth:

```
Back-off restarting failed container database-migrations in pod rafiki-auth-isv7kqx762-7975ff7c6f-gsm9f_rafiki-auth-isv7kqx762(db38dd1f-9266-4a86-8d25-e369d4337462)
```

What happened here is initContainer has started and it tried to ran database migrations. This is what you know.

What you can assume from rafiki repository is that this likely failed because it is using "old" docker image that does not yet have support for manual migrations. 

We will probably not be able to test this before the new rafiki release. With that in mind, I will merge this PR

**Pull request merged**